appservice/src/main/java/nu/yona/server/messaging/rest/Decorator.java

```diff
@@ -0,0 +1,13 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Stichting Yona Foundation This Source Code Form is subject to the terms of the Mozilla Public License, v.
+ * 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *******************************************************************************/
+package nu.yona.server.messaging.rest;
+
+import nu.yona.server.messaging.rest.MessageController.MessageResourceAssembler;
+import nu.yona.server.messaging.service.MessageDto;
+
+public interface Decorator
```

Comment:
I find the namings in this PR confusing. First, we do not really implement the [decorator design pattern](https://www.baeldung.com/java-decorator-pattern). Second, we do use annotations here to decorate classes which are called decorators too. We decorate the decorators...

Proposal:

- Putting the added links of separate message types in separate classes is good for keeping things simple. 
- Using annotations to get the right class looks nice and gives the possibility to implement it for both abstract classes and concrete classes. However, we could also use a type parameter maybe? So use `interface MessageResourceAssembler<T>`?
- Rename `Decorator` -> `MessageResourceAssembler` to avoid confusion about the name
- Rename `DecoratorRegistry` -> `MessageResourceAssemblerRegistry`

What do you think? Maybe it's best to discuss this offline :)

Reply:
Interesting. I didn't think of the GoF decorators at all.
The word "decorate" has many meanings. I continued to use the term "decorate" in line with what we already did, as expressed in the method ``doDynamicDecoration``. The decorators decorate the resource with extra links and/or embedded resources.

You also use the term in two different meanings:
* The GoF decorator pattern
* Java annotations

The "decorators" aren't message resource assemblers, so that name would be wrong. The resource is partially assembled already and the "decorator" is a helper for that.

My first attempt as to make the decorators typed (using ``public interface Decorator<T extends MessageDto>``), but that does not add any value, as the registration is consulted in a typeless way. After quite a struggle, I concluded I couldn't make it work with a type parameter.

So I'm open to a different name for "decorator", to prevent confusion with the GoF pattern, but we can't use ``MessageResourceAssembler``.